Sources/NIO/BaseSocket.swift

```diff
@@ -12,6 +12,28 @@
 //
 //===----------------------------------------------------------------------===//
 
+import NIOConcurrencyHelpers
+
+private let globallyIgnoredSIGPIPE: Atomic<Bool> = .init(value: false)
+
+func ignoreSIGPIPE(descriptor fd: CInt) throws {
+    if !globallyIgnoredSIGPIPE.load() {
+        #if os(Linux)
+        /* no F_SETNOSIGPIPE on Linux :( */
+        _ = unsafeBitCast(Glibc.signal(SIGPIPE, SIG_IGN) as sighandler_t?, to: Int.self)
```

Comment:
What's the point of an `unsafeBitCast` if we're throwing away the result?

Reply:
none :)